src/java.base/share/classes/java/util/zip/ZipCoder.java

```diff
@@ -158,7 +158,7 @@ static int hash(String name) {
         return hsh;
     }
 
-    boolean hasTrailingSlash(byte[] a, int end) {
+    protected boolean hasTrailingSlash(byte[] a, int end) {
```

Comment:
Can you explain the need for this

Reply:
See other comment to Claes: Since UTF8ZipCoder.compare is now the only caller of this method, it can be made private.